pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -621,34 +638,44 @@ pub mod pallet {
 		type Message = GatewayMessage<T::Message, T::RouterId>;
 
 		fn process(msg: Self::Message) -> (DispatchResult, Weight) {
-			match msg {
-				GatewayMessage::Inbound {
-					domain_address,
-					message,
-					router_id,
-				} => {
-					let mut counter = 0;
-
-					let res = Self::process_inbound_message(
+			// The #[transactional] macro only works for functions that return a
+			// `DispatchResult` therefore, we need to manually add this here.
+			with_transaction(|| {
+				let (res, weight) = match msg {
+					GatewayMessage::Inbound {
 						domain_address,
 						message,
 						router_id,
-						&mut counter,
-					);
+					} => {
+						let mut counter = 0;
 
-					let weight = match counter {
-						0 => LP_DEFENSIVE_WEIGHT / 10,
-						n => LP_DEFENSIVE_WEIGHT.saturating_mul(n),
-					};
+						let res = Self::process_inbound_message(
+							domain_address,
+							message,
+							router_id,
+							&mut counter,
+						);
 
-					(res, weight)
-				}
-				GatewayMessage::Outbound { message, router_id } => {
-					let res = T::MessageSender::send(router_id, T::Sender::get(), message);
+						(res, Self::get_weight_for_batch_messages(counter))
```

Comment:
Question: This can also be simplified if no inbound messages are supported right?

Reply:
Yes, we can always return the defensive weight.